### PR TITLE
Remove withCompoent

### PR DESCRIPTION
**Pull request by SaraVieira**
Closes https://trello.com/c/bVAB5XQc/248-replace-withcomponent-calls-with-as-prop-on-styled-components

### CODE REVIEW

File: src/components/Common/CaseStudyCards/CaseStudyHero.js [hunk 1]
```diff
@@ -36,7 +34,7 @@ const SpecialitiesAndServices = ({ specialities, services }) => (
 const RightHandText = ({ title, specialities, services }) => (
   <Fragment>
     <Padding bottom={0.5}>
-      <CardHeading>{title}</CardHeading>
+      <H2>{title}</H2>
```
[the remark this answers is not in the document]
Comment (SaraVieira) [reply src/components/Common/CaseStudyCards/CaseStudyHero.js]: I think we are using the H2 componetent widely enough to deserve it's own compobnent

File: src/components/Typography/h2.js [hunk 2]
```diff
@@ -3,10 +3,16 @@ import remcalc from 'remcalc'
...
 import is from 'styled-is'
-import H1 from './h1'
+import { h1Styles } from './h1'
 
 // this is to be able to pass the className prop down to add styles on later
-const H2 = props => React.createElement(H1.withComponent('h2'), props)
+const H2 = props =>
+  React.createElement(
+    styled.h2`
+      ${h1Styles}
```
Comment (jsms90) [inline src/components/Typography/h2.js]: Nice :+1:

File: src/components/Common/CaseStudyCards/CaseStudyHero.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jsms90) [reply src/components/Common/CaseStudyCards/CaseStudyHero.js]: Oh right. Woops. The H2 component has the same styles as the H1 component. Just saw that.

Yeah cool :+1:

File: src/components/Typography/h2.js [hunk 3]
```diff
@@ -3,10 +3,16 @@ import remcalc from 'remcalc'
 import styled from 'styled-components'
```
Comment (antoniocapelo) [inline src/components/Typography/h2.js]: couldn't we just 

```
<H1 as="h2">{children}</H1>
``` 
?

File: src/components/Typography/h2.js (hunk 2 above)
Comment (SaraVieira) [reply src/components/Typography/h2.js]: 🎉

File: src/components/Common/CaseStudyCards/CaseStudyHero.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (antoniocapelo) [reply src/components/Common/CaseStudyCards/CaseStudyHero.js]: makes sense

File: src/components/Typography/h2.js (hunk 3 above)
Comment (SaraVieira) [reply src/components/Typography/h2.js]: It's the same thing but I find this one buildable upon if we need to add something

Comment (SaraVieira) [reply src/components/Typography/h2.js]: I updated the h2

**Pull request merged**